cupy/random/_generator.py

```diff
@@ -1150,26 +1150,51 @@ def randint(self, low, high=None, size=None, dtype=int):
             - :func:`cupy.random.randint` for full documentation
             - :meth:`numpy.random.RandomState.randint`
         """
-        if high is None:
-            lo = 0
-            hi1 = int(low) - 1
+        if not (numpy.isscalar(low) and numpy.isscalar(high)):
+            low = cupy.asarray(low)
+            if high is None:
+                lo = cupy.zeros_like(low)
+                hi = low - 1
+            else:
+                lo = low
+                hi = cupy.asarray(high) - 1
+
+            if cupy.any(lo > high):
+                raise ValueError('low >= high')
+
```

Comment:
This check causes device to host sync, which may harm its performance. We could remove this check and make it the user's responsibility as `uniform` does.